File: frontend/src/app/shared/dto/classroom.dto.ts
export class ClassroomDto {
  public readonly id: number | null;
  public readonly name: string;
  public readonly subject: string;
  public readonly commitmentPeriod: boolean;
  public readonly archived: boolean;
  public readonly modulIds: number[];
  public readonly applicationUserIds: number[];

  constructor(
    id: number | null,
    name: string,
    subject: string,
    commitmentPeriod: boolean,
    archived: boolean,
    modulIds: number[],
    applicationUserIds: number[]
  ) {
    this.id = id;
    this.name = name;
    this.subject = subject;
    this.commitmentPeriod = commitmentPeriod;
    this.archived = archived;
    this.modulIds = modulIds;
    this.applicationUserIds = applicationUserIds;
  }
}

File: backend/src/main/java/com/example/backend/model/dto/ClassroomDTO.java
package com.example.backend.model.dto;

import java.util.Set;

public class ClassroomDTO {
    private Long id;
    private String name;
    private String subject;
    private Boolean commitmentPeriod;
    private Boolean archived;
    private Set<Long> modulIds;
    private Set<Long> applicationUserIds;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public Boolean getArchived() {
        return archived;
    }

    public void setArchived(Boolean archived) {
        this.archived = archived;
    }

    public Set<Long> getModulIds() {
        return modulIds;
    }

    public void setModulIds(Set<Long> modulIds) {
        this.modulIds = modulIds;
    }

    public Set<Long> getApplicationUserIds() {
        return applicationUserIds;
    }

    public void setApplicationUserIds(Set<Long> applicationUserIds) {
        this.applicationUserIds = applicationUserIds;
    }

    public Boolean getCommitmentPeriod() {
        return commitmentPeriod;
    }

    public void setCommitmentPeriod(Boolean commitmentPeriod) {
        this.commitmentPeriod = commitmentPeriod;
    }
}
